x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -470,7 +491,34 @@ void searchVersionedDocumentForSubject(Subject subject, ActionListener<Versioned
         });
     }
 
-    private void createNewProfile(Subject subject, String uid, ActionListener<Profile> listener) throws IOException {
+    private static final String INVALID_USERNAME_MESSAGE = "Security domain [%s] is configured to use literal username. "
+        + "As a result, creating new user profile requires the username to be at least 1 and no more than 490 characters. "
+        + "The username can contain alphanumeric characters (a-z, A-Z, 0-9), spaces, punctuation, "
+        + "and printable symbols in the Basic Latin (ASCII) block.";
+
+    private ValidationException validateUsername(Subject subject) {
+        final RealmDomain realmDomain = subject.getRealm().getDomain();
+        assert realmDomain != null;
+        assert domainConfigLookup.apply(realmDomain.name()) != null;
+        assert domainConfigLookup.apply(realmDomain.name()).literalUsername();
+
+        final String username = subject.getUser().principal();
+        assert username != null;
+
+        if (username.length() < 1 || username.length() > 490) {
```

Comment:
The value `490` is somewhat artificial. Strictly speaking it is `491` when you take out the prefix (`profile_u_`) and the suffix (`_xx`). But `491` is rather an odd number. So I rounded it to `490`. It is still odd, but I don't want to contraint it to something like 100 without good reasons.

Reply:
This doesn't leave us with any room to change the profile_id generation in the future (for this generation scheme).
I don't think it is important, though I would've made it 256.

Reply:
I think you have a point here. I changed it to 256. Thanks!